### PR TITLE
logging: Replace zap logger by klog

### PR DESCRIPTION
**Pull request by osmman**
This PR makes the following changes:

- Replaced Zap with klog: The zap logging library has been replaced by klog to align with Kubernetes native logging practices. [1]

- Removed Redundant Logs in Reconcile Methods: Unnecessary logging in the Reconcile methods has been removed as comprehensive logging is already available in the controller class. Detailed reconciliation logs can still be activated with verbosity level 5 `-v=5`.

[1] https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md#logging-conventions

### CODE REVIEW

**Comment by osmman** [comment]
you can see a new format of logging in dumps of operator logs in [Github Actions](https://github.com/securesign/secure-sign-operator/actions/runs/10181073289/job/28160812843?pr=533#step:16:1)

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -3,6 +3,6 @@ resources:
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
-- digest: sha256:a21f7128694a64989bf0d84a7a7da4c1ffc89edf62d594dc8bea7bcfe9ac08d3
-  name: controller
-  newName: registry.redhat.io/rhtas/rhtas-rhel9-operator
+- name: controller
```
Comment (bouskaJ) [inline config/manager/kustomization.yaml:6]: Do not replace image

Comment (osmman) [reply config/manager/kustomization.yaml]: fixed

**Comment by bouskaJ** [comment]
Please rebase.
/lgtm

**Pull request merged**